src/react/workflow/ExpirationForm.tsx

```diff
@@ -261,6 +274,27 @@ export function ExpirationForm({ bucketList, locations, prefix = '' }: Props) {
                   </T.ErrorContainer>
                 </T.Value>
               </T.Row>
+              <T.Row style={{ maxHeight: 'initial', height: '100%', marginTop: '2rem', alignItems: 'baseline'}}>
+                <T.Key>Tags</T.Key>
+                <T.Value>
+                  <TagsFilter
+                    {...register(`${prefix}filter.objectTags`)}
+                    handleChange={(data) => { setValue(`${prefix}filter.objectTags`, data, { shouldDirty: true }); }}
```

Comment:
I'm unsure about this approach of spreading register and then in // controlling the change through a handleChange. Usually for such custom component the "Controller" pattern is preferred (https://react-hook-form.com/api/usecontroller/controller/#main). Also you likely don't want to force dirty here because if the user manually reset the form, then the form shouldn't be dirty anymore.